integ-test/src/test/java/org/opensearch/sql/sql/NestedIT.java

```diff
@@ -5,24 +5,52 @@
 
 package org.opensearch.sql.sql;
 
+import static org.opensearch.sql.legacy.TestsConstants.TEST_INDEX_MULTI_NESTED;
+import static org.opensearch.sql.util.MatcherUtils.rows;
+import static org.opensearch.sql.util.MatcherUtils.verifyDataRows;
+
+import org.json.JSONArray;
 import org.json.JSONObject;
 import org.junit.Test;
 import org.opensearch.sql.legacy.SQLIntegTestCase;
 
 import java.io.IOException;
+import java.util.List;
 
 import static org.opensearch.sql.legacy.TestsConstants.TEST_INDEX_NESTED_TYPE;
 
 public class NestedIT extends SQLIntegTestCase {
   @Override
   public void init() throws IOException {
     loadIndex(Index.NESTED);
+    loadIndex(Index.MULTI_NESTED);
+  }
+
+  @Test
+  public void nested_function_with_array_of_nested_field_test() {
+    String query = "SELECT nested(message.info), nested(comment.data) FROM " + TEST_INDEX_NESTED_TYPE;
```

Comment:
I believe our solution is to split the `nested` function for JDBC or other queries that have strict typing (really, any SQL language).  For now, we should make sure that the JDBC and ODBC drivers should not output any undesired typing and break.  

To ensure that typings aren't broken, we could introduce partiql like syntax for nested that would make sure that only strict typing is followed on output.  For example: 

```
SELECT n.name, m1.info AS info_keyword, m2.info AS info_array FROM nested as n, n.message AS m1 IS keyword, n.message as m2 IS array

SELECT n.name, CASE(m.info IS keyword) THEN m.info END AS info_keyword, CASE(m.info IS ARRAY) THEN m.info END AS info_array FROM nested as n, n.message AS m
```

For now, this can be marked as 'out of scope' as the user should be responsible for proper data setup and typing.

Reply:
I agree. It is a new feature reported in [#1300](https://github.com/opensearch-project/sql/issues/1300).